R/test_wrapper.R

```diff
@@ -2134,3 +2157,63 @@ tidy.shapiro_exploratory <- function(x, type = "model", signif_level=0.05) {
     ret
   }
 }
+
+#' @export
+exp_chisq_power <- function(dummy, rows=2, cols=2, w=0.3, sig.level=0.05, beta=0.2, n_start=10, n_end=100, n_step=10) {
+  power <- 1.0 - beta
+  n = seq(n_start, n_end, by=n_step)
+  # t-test
+  # n_to_power_res <- pwr::pwr.t.test(n=n, d=0.3, sig.level=0.05, type="two.sample")
+
+  chisq_power_each <- function(dummy) {
+    df = (rows-1)*(cols-1) # Degree of freedom
+
+    # Sample size vs power calculation
+    n_to_power_res <- pwr::pwr.chisq.test(df=df, N=n, w=w, sig.level=sig.level)
+    n_to_power <- tibble::tibble(n=n, power = n_to_power_res$power)
+
+    # Required sample size calculation
+    required_n <- (pwr::pwr.chisq.test(df=df, N=NULL, w=w, sig.level=sig.level, power=power))$N
+
+    crit <- qchisq(1-sig.level, df=df) # The chisq value that corresponds to the significance level.
+    density <- generate_chisq_density_data_for_power(df=df, w=w, N=required_n, crit)
+
+    model <- list(n_to_power=n_to_power,
+                  df=df,
+                  w=w,
+                  sig.level=sig.level,
+                  beta=beta,
+                  power=power,
+                  required_n=required_n,
+                  density=density)
+    class(model) <- c("chisq_power_exploratory")
+    model
+  }
+  do_on_each_group(dummy, chisq_power_each, name = "model", with_unnest = FALSE)
+}
+
+#' @export
```

Comment:
Add a comment. Describe the difference from `exp_chisq_power`.

Reply:
Done.